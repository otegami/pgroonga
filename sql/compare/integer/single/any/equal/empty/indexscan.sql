CREATE TABLE ids (
  id integer
);

INSERT INTO ids VALUES (1);
INSERT INTO ids VALUES (2);
INSERT INTO ids VALUES (3);

CREATE INDEX pgroonga_index ON ids USING pgroonga (id);

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::integer[])
 ORDER BY id ASC;

SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::integer[])
 ORDER BY id ASC;

DROP TABLE ids;
