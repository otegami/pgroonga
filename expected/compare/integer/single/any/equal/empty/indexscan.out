CREATE TABLE ids (
  id integer
);
INSERT INTO ids VALUES (1);
INSERT INTO ids VALUES (2);
INSERT INTO ids VALUES (3);
CREATE INDEX pgroonga_index ON ids USING pgroonga (id);
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::integer[])
 ORDER BY id ASC;
                 QUERY PLAN                 
--------------------------------------------
 Index Scan using pgroonga_index on ids
   Index Cond: (id = ANY ('{}'::integer[]))
(2 rows)

SELECT id
  FROM ids
 WHERE id = ANY(ARRAY[]::integer[])
 ORDER BY id ASC;
 id 
----
(0 rows)

DROP TABLE ids;
